src/androidtvremote2/remote.py

```
@@ -237,3 +331,31 @@ async def _async_idle_disconnect(self) -> None:
             self.transport.close()
         if not self.on_con_lost.done():
             self.on_con_lost.set_result(Exception("Closed idle connection"))
+
+    async def _async_wait_for_future_or_con_lost(self, future: asyncio.Future[Any], timeout: float) -> Any:
+        """Wait for the future to finish, connection to be lost, or timeout occurs.
+
+        :param future: The future to wait for.
+        :param timeout: Timeout in seconds.
+
+        :raises ConnectionClosed: If the connection is lost or the future has an exception.
+        :raises asyncio.TimeoutError: If timeout is reached before completion.
+        """
+        tasks = {self.on_con_lost, future}
```

If the on_con_lost future completes with an exception, this exception is not propagated and will result in a generic ConnectionClosed being raised.